algosdk/abi/interface.py

```diff
@@ -0,0 +1,39 @@
+import json
+
+from algosdk.abi.method import Method
+
+
+class Interface:
+    """
+    Represents a ABI interface description.
+
+    Args:
+        name (string): name of the interface
+        methods (list): list of Method objects
+    """
+
+    def __init__(self, name, methods) -> None:
+        self.name = name
+        self.methods = methods
+
+    def __eq__(self, o: object) -> bool:
+        if not isinstance(o, Interface):
+            return False
+        return self.name == o.name and self.methods == o.methods
+
+    @staticmethod
```

Comment:
This, as well as, `undictify()`, is a good candidate to become a `@classmethod`. That way you could write more generic versions that are a bit more robust to copy-pasta and refactorings. EG (assuming no type hints):

```python
@classmethod
def from_json(cls, resp):
    return cls.undictify(json.loads(resp))

...

@classmethod
def undictify(cls, resp):
    name = d["name"]
    method_list = [Method.undictify(method) for method in d["methods"]]
    return cls(name=name, methods=method_list)
```

Reply:
Good point, I think I will leave it as is for this PR - the existing `dictify` and `undictify` methods (in `Transaction` for instance) follow this so we can consider refactoring this in a later PR all at once.